### PR TITLE
Add program counter, fix types issue

### PR DESCRIPTION
**Pull request by gooxpf**
# Breaking Changes
Create Program Counter

# Changes

- feat: Add signAndSendTransaction method to the WC and Phantom
- fix: sendTransaction and store connection

### CODE REVIEW

File: packages/solana/src/connectors/walletConnectConnector.ts [hunk 1]
```diff
@@ -144,16 +146,56 @@ export class WalletConnectConnector extends BaseConnector implements Connector {
...
+    return signature
+  }
 
-    return base58.encode(signedTransaction)
+  public async signAndSendTransaction(
+    transactionParam: Transaction | VersionedTransaction,
+    signers: Signer[]
+  ) {
+    if (transactionParam instanceof VersionedTransaction) {
+      return 'it doesnt work with versioned transaction'
```
Comment (enesozturk) [inline packages/solana/src/connectors/walletConnectConnector.ts:176]: Should this throw error or return string? How will this be handled on the higher level?

File: packages/solana/src/connectors/walletAdapters/PhantomAdapter.ts [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    transactionParam: Transaction | VersionedTransaction,
+    signers: Signer[],
+    confirmOptions?: ConfirmOptions
+  ) {
+    if (!SolStoreUtil.state.connection) {
+      throw Error('Not Connected')
+    }
+
+    if (transactionParam instanceof VersionedTransaction) {
+      return 'it doesnt work with versioned transaction'
```
Comment (enesozturk) [inline packages/solana/src/connectors/walletAdapters/PhantomAdapter.ts:19]: Same

File: apps/laboratory/src/components/Solana/SolanaWriteContractTest.tsx [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+  const toast = useToast()
+  const { address, currentChain } = useWeb3ModalAccount()
+  const { walletProvider, connection } = useWeb3ModalProvider()
+  const [loading, setLoading] = useState(false)
+
+  async function onIncrementCounter() {
+    setLoading(true)
+
+    const PROGRAM_ID = new PublicKey(detectProgramId(currentChain?.chainId ?? ''))
+    /* Add your program id */
```
Comment (enesozturk) [inline apps/laboratory/src/components/Solana/SolanaWriteContractTest.tsx:27]: Do we need this comment?

Comment (gooxpf) [reply apps/laboratory/src/components/Solana/SolanaWriteContractTest.tsx]: I guess nope.

File: packages/solana/src/connectors/walletConnectConnector.ts (hunk 1 above)
Comment (gooxpf) [reply packages/solana/src/connectors/walletConnectConnector.ts]: Changed to throw Error

File: packages/solana/src/connectors/walletConnectConnector.ts [hunk 4]
```diff
@@ -144,16 +146,56 @@ export class WalletConnectConnector extends BaseConnector implements Connector {
...
   }
 
-  public async sendTransaction(transactionParam: Transaction | VersionedTransaction) {
+  private async _sendTransaction(transactionParam: Transaction | VersionedTransaction) {
     const encodedTransaction = (await this.signTransaction(transactionParam)) as {
       signatures: {
         signature: string
       }[]
     }
+    // Const signedTransaction = transactionParam.serialize()
```
Comment (enesozturk) [inline packages/solana/src/connectors/walletConnectConnector.ts:155]: What is this comment for? Is it something we'd like developers to see? If yes, could we add JSDoc?

File: packages/solana/src/connectors/walletConnectConnector.ts [hunk 5]
```diff
@@ -144,16 +146,56 @@ export class WalletConnectConnector extends BaseConnector implements Connector {
     return { signatures: [{ signature: base58.encode(transaction.serialize()) }] }
   }
 
-  public async sendTransaction(transactionParam: Transaction | VersionedTransaction) {
+  private async _sendTransaction(transactionParam: Transaction | VersionedTransaction) {
```
[the remark this answers is not in the document]
Comment (enesozturk) [reply packages/solana/src/connectors/walletConnectConnector.ts:149]: +1

File: packages/solana/src/connectors/walletAdapters/PhantomAdapter.ts [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+    transactionParam: Transaction | VersionedTransaction,
+    signers: Signer[],
+    confirmOptions?: ConfirmOptions
+  ) {
+    if (!SolStoreUtil.state.connection) {
+      throw Error('Not Connected')
+    }
+
+    if (transactionParam instanceof VersionedTransaction) {
+      throw Error('it doesnt work with versioned transaction')
```
Comment (tomiir) [inline packages/solana/src/connectors/walletAdapters/PhantomAdapter.ts:19]: I'd rephrase this to something like "Versioned transactions are not supported"

File: apps/laboratory/src/components/Solana/SolanaWriteContractTest.tsx [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+          LAMPORTS_PER_SOL
+        )
+
+        const latestBlockHash = await connection.getLatestBlockhash()
+        await airdropConnection.confirmTransaction({
+          blockhash: latestBlockHash.blockhash,
+          lastValidBlockHeight: latestBlockHash.lastValidBlockHeight,
+          signature
+        })
+        throw Error('Not enough SOL in wallet')
```
Comment (tomiir) [inline apps/laboratory/src/components/Solana/SolanaWriteContractTest.tsx:54]: IIUC you are airdropping tokens to the wallet right? if so, why throw here?

File: packages/solana/src/connectors/walletConnectConnector.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (gooxpf) [reply packages/solana/src/connectors/walletConnectConnector.ts:149]: The reason is because in wallet adapters [sendTransaction](https://github.com/anza-xyz/wallet-adapter/blob/master/packages/wallets/phantom/src/adapter.ts) returns only signature, so i created  _sendTransaction here which returns tx and signuture, but public sendTransaction has same type as in wallet adapters.

**Review by svenvoskamp — APPROVED** [review]

**Review by enesozturk — APPROVED** [review]
